Prog/LgUnit - Ipums Table 2B.sas

```diff
@@ -0,0 +1,543 @@
+
+	/* Moved-in */
+	%if &yrs. = 2000 %then %do;
+
+	if movedin in (1) then movedless1 = 1;
+		else movedless1 = 0;
+
+	if movedin in (2,3,4,5) then moved2to10 = 1;
+		else moved2to10 = 0;
+
+	if movedin in (6,7,8) then moved10plus = 1;
+		else moved10plus = 0;
+
+	%end;
+
+	%else %if &yrs. = ACS %then %do;
+
+	if movedin in (1) then movedless1 = 1;
+		else movedless1 = 0;
+
+	if movedin in (2,3,4) then moved2to10 = 1;
+		else moved2to10 = 0;
+
+	if movedin in (5,6,7,8) then moved10plus = 1;
+		else moved10plus = 0;
+
+	%end; 
+
+	/* Bedroom size */
+	if bedrooms = 1 then bedrooms0 = 1;
+		else bedrooms0 = 0;
+
+	if bedrooms = 2 then bedrooms1 = 1;
+		else bedrooms1 = 0;
+
+	if bedrooms = 3 then bedrooms2 = 1;
+		else bedrooms2 = 0;
+
+	if bedrooms = 4 then bedrooms3 = 1;
+		else bedrooms3 = 0;
+
+	if bedrooms = 5 then bedrooms4 = 1;
+		else bedrooms4 = 0;
+
+	if bedrooms > 5 then bedrooms5plus = 1;
+		else bedrooms5plus = 0;
+
+	/* Units in structure */
+	if unitsstr in (3,4) then units1 = 1; 
+		else units1 = 0;
+
+	if unitsstr in (5,6) then units2to4 = 1;
+		else units2to4 = 0;
+
+	if unitsstr in (7) then units5to9 = 1;
+		else units5to9 = 0;
+
+	if unitsstr in (8) then units10to19 = 1;
+		else units10to19 = 0;
+
+	if unitsstr in (9,10) then units20plus = 1;
+		else units20plus = 0;
+
+	/* Year built */
+	if builtyr2 = 1 then BuiltBefore1940 = 1; 
+		else BuiltBefore1940 = 0;
+
+	if builtyr2 in (2,3) then Built1940to1959 = 1;
+		else Built1940to1959 = 0;
+
+	if builtyr2 in (4,5) then Built1960to1979 = 1;
+		else Built1960to1979 = 0;
+
+	if builtyr2 in (6,7) then Built1980to1999 = 1;
+		else Built1980to1999 = 0;
+
+	if builtyr2 >= 9 then Built2000After = 1;
+		else Built2000After = 0;
+
+	keep &cvars. &mvars. nonrelative numkids numadults
+		  serial hhwt pernum numprec race hispan age hhincome;
+  
+run;
+
+
+proc summary data = pretables;
+	class serial;
+	var &cvars. nonrelative numkids numadults;
+	output out= pretables_s sum=;
+run;
+
+
+proc sort data = pretables_s; by serial; run;
+
+
+proc summary data = pretables;
+	class serial;
+	var &mvars.;
+	output out= pretables_m median=;
+run;
+
+proc sort data = pretables_m; by serial; run;
+
+
+data pretables_collapse;
+	merge pretables_s  pretables_m hhwts;
+	by serial;
+	if serial ^= .;
+
+	%macro onezero();
+		%let varlist = &cvars.;
+			%let i = 1;
+				%do %until (%scan(&varlist,&i,' ')=);
+					%let var=%scan(&varlist,&i,' ');
+			if &var. >= 1 then &var. = 1;
+		%let i=%eval(&i + 1);
+				%end;
+			%let i = 1;
+				%do %until (%scan(&varlist,&i,' ')=);
+					%let var=%scan(&varlist,&i,' ');
+		%let i=%eval(&i + 1);
+				%end;
+	%mend onezero;
+	%onezero;
+
+	/* Flag 3+ generation households */
+	numgens = sum(of before1gen before2gen after1gen);
+	if numgens >=2 then multigen = 1;
+		else multigen = 0;
+
+	/* Flag "group houses" */
+	if nonrelative >= 3 then grouphouse = 1;
+		else grouphouse = 0;
+
+
+run;
+
+
+/* Add vacancy data */
+data pretables_withvac;
+	set pretables_collapse 
```

Comment:
keep the other vars we need? rent, valueh, unistr, etc. ?

Reply:
Decided we didn't need a keep statement here. Might as well just keep all the vars.